Xamarin.Forms.Core/NavigationPage.cs

```diff
@@ -319,7 +319,8 @@ void InsertPageBefore(Page page, Page before)
 
 			var root = (Page)PageController.InternalChildren.First();
 
-			PageController.InternalChildren.ToArray().Where(c => c != root).ForEach(c => PageController.InternalChildren.Remove(c));
+            var childrenToRemove = PageController.InternalChildren.ToArray().Where(c => c != root);  
+            childrenToRemove.ForEach(c => PageController.InternalChildren.Remove(c)); 
```

Comment:
Could you replace this by a real `foreach` loop ? We're trying to avoid linq extensions method as much as we can, as they often generate unnecessary allocation load.